config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleMetadataBuilder.java

```diff
@@ -63,7 +63,15 @@ public BundleMetadata build(final AnnotatedBundle annotatedBundle, final Bundle
             builder.dependencies(annotatedBundle.getDependentBundles());
             builder.tags(annotatedEntity.getTags());
             builder.redeployable(annotatedEntity.isRedeployable() || !isBundleContainsReusableEntity(annotatedBundle));
-            builder.l7Template(Boolean.valueOf(String.valueOf(encass.getProperties().get(L7_TEMPLATE))));
+            boolean l7Template = false;
+            switch (annotatedEntity.getEntityType()) {
+                case EntityTypes.ENCAPSULATED_ASSERTION_TYPE:
+                    l7Template = Boolean.valueOf(String.valueOf(((Encass) annotatedEntity.getEntity()).getProperties().get(L7_TEMPLATE)));
```

Comment:
can we move this logic to AnnotatedEnity class so that this class we be free EntityType checks

Reply:
done